libursa/src/encryption/symm/aescbc.rs

```diff
@@ -0,0 +1,141 @@
+use super::Encryptor;
+use aead::{
+    generic_array::{
+        typenum::{Unsigned, U0, U16, U32, U48, U64},
+        GenericArray,
+    },
+    Aead, Error, NewAead, Payload,
+};
+use aes::{Aes128, Aes256};
+use block_modes::block_padding::Pkcs7;
+use block_modes::{BlockMode, Cbc};
+use hmac::{Hmac, Mac};
+#[cfg(feature = "serialization")]
+use serde::de::{Deserialize, Deserializer, Error as DError, Visitor};
+#[cfg(feature = "serialization")]
+use serde::ser::{Serialize, Serializer};
+use sha2::{Sha256, Sha512};
+use subtle::ConstantTimeEq;
+use zeroize::Zeroize;
+
+type Aes128Cbc = Cbc<Aes128, Pkcs7>;
+type Aes256Cbc = Cbc<Aes256, Pkcs7>;
+type HmacSha256 = Hmac<Sha256>;
+type HmacSha512 = Hmac<Sha512>;
+
+macro_rules! aes_cbc_hmac_impl {
+    ($name:ident, $algokeysize:ident, $keysize:ident, $noncesize:ident, $tagsize:ident, $algo:ident, $hash:ident, $visitor:ident) => {
+        #[derive(Debug, Clone, Eq, PartialEq)]
+        pub struct $name {
+            key: GenericArray<u8, $keysize>,
+        }
+
+        impl Encryptor for $name {
+            type MinSize = U48;
```

Comment:
Is `MinMinSize` always `U48`?

Reply:
Read the comments in https://github.com/hyperledger/ursa/pull/76/files#diff-882b2ef604a9130ce242dd63b5458972R213

Reply:
Each algorithm has a min size thats fixed.

Reply:
See other comments.